File: backend/model/forensics.go
package model

import "time"

// ForensicTask - (Case/沙箱)
type ForensicTask struct {
	ID          uint           `json:"id" gorm:"primarykey"`
	Name        string         `json:"name" gorm:"type:varchar(255);not null"`
	Description string         `json:"description" gorm:"type:text"`
	Status      string         `json:"status" gorm:"type:varchar(50);default:'open'"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	
	//   ✅ New增：一对多关联，方便Query时一次性拉出所有EvidenceFile
	Files       []ForensicFile `json:"files" gorm:"foreignKey:TaskID"` 
}

// ForensicFile - (保持你原来的不变即可)
type ForensicFile struct {
	ID           uint      `json:"id" gorm:"primarykey"`
	TaskID       uint      `json:"task_id" gorm:"index;not null"`
	FileName     string    `json:"file_name" gorm:"type:varchar(255);not null"`
	OriginalName string    `json:"original_name" gorm:"type:varchar(255)"`
	FileType     string    `json:"file_type" gorm:"type:varchar(50)"` 
	FileSize     int64     `json:"file_size"`
	FilePath     string    `json:"file_path" gorm:"type:varchar(500)"`                   
	ParseStatus  string    `json:"parse_status" gorm:"type:varchar(50);default:'pending'"` 
	ParseMessage string    `json:"parse_message" gorm:"type:text"`                         
	EventCount   int       `json:"event_count"`                                            
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

File: frontend/src/services/forensics.ts
import { apiClient } from "@/lib/api/vsentry-client";
import type { APIResponse } from "@/lib/api/vsentry-client";

// ==================== Data模型Type定义 ====================

export type ParseStatus = 'pending' | 'parsing' | 'completed' | 'failed';

export interface ForensicFile {
  id: number;
  task_id: number;
  file_name: string;
  original_name: string;
  file_type: string;
  file_size: number;
  file_path: string;
  parse_status: ParseStatus;
  parse_message: string;
  event_count: number;
  created_at: string;
  updated_at: string;
}

export interface ForensicTask {
  id: number;
  name: string;
  description: string;
  status: 'open' | 'closed';
  created_at: string;
  updated_at: string;
  // 后端 Preload 关联带出的FileList
  files?: ForensicFile[]; 
}

// ==================== API 交互层 ====================

export const forensicsService = {
  // 1. Case (Task) Manage
  listTasks: () => 
    apiClient.get<any, APIResponse<ForensicTask[]>>("/forensics/tasks"),
    
  createTask: (data: { name: string; description?: string }) => 
    apiClient.post<any, APIResponse<ForensicTask>>("/forensics/tasks", data),
    
  getTask: (id: number | string) => 
    apiClient.get<any, APIResponse<ForensicTask>>(`/forensics/tasks/${id}`),
    
  deleteTask: (id: number | string) => 
    apiClient.delete<any, APIResponse<any>>(`/forensics/tasks/${id}`),

  // 2. EvidenceFile (File) Manage
  uploadFile: (taskId: number | string, file: File) => {
    const formData = new FormData();
    formData.append("task_id", String(taskId));
    formData.append("file", file);

    const timeout = file.size > 50 * 1024 * 1024 ? 300000 : 60000;

    return apiClient.post<any, APIResponse<ForensicFile>>("/forensics/upload", formData, {
      timeout,
      headers: {
        "Content-Type": "multipart/form-data",
      },
    });
  },

  deleteFile: (id: number | string) => 
    apiClient.delete<any, APIResponse<any>>(`/forensics/files/${id}`),

  // 3. ExecuteForensicsRule
  executeRules: (caseId: number, fileId: number, ruleIds: number[]) => 
    apiClient.post<any, APIResponse<any>>("/forensics/execute-rules", {
      case_id: caseId,
      file_id: fileId,
      rule_ids: ruleIds,
    }),
};